Create monitoring-guide.md (#20999)
sourcegraph/sourcegraph: doc/admin/how-to/monitoring-plan.md

# Monitoring Guide
Please visit our [Observability Docs](./observability) for more in-depth information about observability.

## What should I be monitoring?

> Sourcegraph comes with built-in monitoring in the form of [Grafana](./observability/metrics#grafana) using [Prometheus](./observability/metrics#prometheus) for metrics and alerting. Generally, you can access Grafana by visiting `https://<your-sourcegraph-url>/-/debug/grafana` for monitoring purpose.

Reviewer: ```suggestion
Sourcegraph comes with built-in monitoring in the form of [Grafana](./observability/metrics#grafana) using [Prometheus](./observability/metrics#prometheus) for metrics and alerting. Generally, you can access Grafana by visiting `https://<your-sourcegraph-url>/-/debug/grafana` for monitoring purposes.
```

## When should I be monitoring?

> Whenever you see an alert on the Grafana dashboard.

Reviewer: This section is a bit fuzzy. What does the act of monitoring mean? Could this be rephrased as "What should I look at when I receive an alert?"

Author: That's a good call. I have updated the section with more information about what to look for when experiencing a system performance issue.

learn how they are defined. Warnings are typically not as important as critical but should be investigated. 
All alerts turn red when they fire. Critical alerts should be investigated and reported if they are occurring repeatedly.

## How do I know when more resources is needed for a specified service?
All resources contain a dashboard called `Provisioning indicators` that provide information about the current resource usage of containers. These can be used to determine if a scale-up is needed.

## What does this <ALERT-MESSAGE> mean?
See [Alert solutions](https://docs.sourcegraph.com/admin/observability/alert_solutions) to learn about each alert and their possible solutions. 

## What’s the threshold for each resource?
All resources have provisioning indicators that will fire up alerts when the container uses 80% of the container memory limit and 90% of the specified CPU limit.

## How much resources should I add after receiving alerts about running out of resources?
You should make the decision based on the metrics from your Grafana Dashboard. 

> ## What are some of the important alerts that I should beaware of?

Reviewer: ```suggestion
## What are some of the important alerts that I should be aware of?
```

> All alerts are important, but more important when they are in red as that indicate there are issues occuring continuously that require attention.

Reviewer: Better to recommend focusing only on critical alerts for now.

Author: Update as suggested!

## How to set up alerts?
See our Observability Docs on [setting up alerting](./observability/alerting#setting-up-alerting).

## How to create a custom alert?

> Creating a customer alert is not recommended and currently not supported by Sourcegraph.

Reviewer: ```suggestion
Creating a customer alert is not recommended and currently not supported by Sourcegraph. However, please provide feedback on the monitoring dashboards and alerts if you find anything could be improved.
```
